lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -8,17 +8,20 @@ const HarmonyExportImportedSpecifierDependency = require("../dependencies/Harmon
 const HarmonyImportSideEffectDependency = require("../dependencies/HarmonyImportSideEffectDependency");
 const HarmonyImportSpecifierDependency = require("../dependencies/HarmonyImportSpecifierDependency");
 
+const isSideEffectFree = resolveData => {
```

Comment:
I didn't see the point in burying this function in the class, so it's outside.  I'm not sure if this matches the code styles well.